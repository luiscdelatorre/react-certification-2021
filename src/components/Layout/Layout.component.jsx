import React from 'react';
import {
  IoEnterOutline,
  IoHomeOutline,
  IoOptions,
  IoPersonCircle,
  IoSearchOutline,
  IoStar,
} from 'react-icons/io5';
import {
  Container,
  Header,
  Menu,
  MenuList,
  MenuItem,
  HeaderLeft,
  HeaderRight,
  HeaderCenter,
  MainContent,
  MenuIcon,
  Search,
  SearchInput,
  User,
} from './Layout.styles';

function Layout({ children }) {
  return (
    <Container>
      <Header>
        <HeaderLeft>
          <h3>Brand</h3>
        </HeaderLeft>
        <HeaderCenter>
          <Search>
            <IoSearchOutline />
            <SearchInput type="text" placeholder="Search" />
          </Search>
        </HeaderCenter>
        <HeaderRight>
          <User>
            <IoPersonCircle />
          </User>
        </HeaderRight>
      </Header>
      <Menu>
        <MenuList>
          <MenuItem active>
            <MenuIcon>
              <IoHomeOutline />
              <small>Home</small>
            </MenuIcon>
          </MenuItem>
          <MenuItem>
            <MenuIcon>
              <IoStar />
              <small>Favorites</small>
            </MenuIcon>
          </MenuItem>
          <MenuItem>
            <MenuIcon>
              <IoOptions />
              <small>Options</small>
            </MenuIcon>
          </MenuItem>
          <MenuItem>
            <MenuIcon>
              <IoEnterOutline />
              <small>Login</small>
            </MenuIcon>
          </MenuItem>
        </MenuList>
      </Menu>
      <MainContent>{children}</MainContent>
    </Container>
  );
}

export default Layout;
