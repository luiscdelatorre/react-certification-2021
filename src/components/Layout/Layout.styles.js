import styled from 'styled-components';
import device from '../../config/device';

const Container = styled.div`
  width: 100vw;
  height: 100vh;
  display: grid;
  grid-template-areas:
    'header'
    'content'
    'menu';
  grid-template-rows: 8rem auto 7rem;

  @media ${device.laptop} {
    grid-template-areas:
      'header header'
      'menu content';
    grid-template-rows: 9rem auto;
    grid-template-columns: ${(props) => (props.isMenuExpanded ? '20rem' : '9rem')} auto;
  }
`;

const MainContent = styled.main`
  grid-area: content;
  background-color: ${(props) => props.theme.background};
  overflow-y: auto;
`;

const Styled = { Container, MainContent };
export default Styled;
