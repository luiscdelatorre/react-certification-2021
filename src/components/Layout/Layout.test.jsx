import { render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AuthProvider from '../../providers/Auth';
import SearchProvider from '../../providers/Search';
import Layout from './Layout.component';

describe('Layout component', () => {
  it('Should check main components in Layout', () => {
    render(
      <BrowserRouter>
        <AuthProvider>
          <SearchProvider>
            <Layout />
          </SearchProvider>
        </AuthProvider>
      </BrowserRouter>
    );

    expect(screen.queryByTestId('header')).toBeInTheDocument();
    expect(screen.queryByTestId('menu')).toBeInTheDocument();
    expect(screen.queryByTestId('main-content')).toBeInTheDocument();
  });

  it('Validates Layout snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <AuthProvider>
          <SearchProvider>
            <Layout />
          </SearchProvider>
        </AuthProvider>
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
