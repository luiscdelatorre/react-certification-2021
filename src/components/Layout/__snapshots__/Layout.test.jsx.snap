// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component Validates Layout snapshot 1`] = `
<div
  className="sc-bdfBwQ iGzSrb"
>
  <header
    className="sc-gsTCUz bASDNS"
  >
    <div
      className="sc-dlfnbm sc-hKgILt mMVyH bEbniD"
    >
      <h3>
        Brand
      </h3>
    </div>
    <div
      className="sc-dlfnbm sc-eCssSg mMVyH beFDfs"
    >
      <div
        className="sc-dlfnbm sc-gKsewC mMVyH kLoGAS"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M221.09 64a157.09 157.09 0 10157.09 157.09A157.1 157.1 0 00221.09 64z"
            fill="none"
            strokeMiterlimit="10"
            strokeWidth="32"
          />
          <path
            d="M338.29 338.29L448 448"
            fill="none"
            strokeLinecap="round"
            strokeMiterlimit="10"
            strokeWidth="32"
          />
        </svg>
        <input
          className="sc-iBPRYJ cA-djgn"
          placeholder="Search"
          type="text"
        />
      </div>
    </div>
    <div
      className="sc-dlfnbm sc-jSgupP mMVyH bbslsP"
    >
      <div
        className="sc-dlfnbm sc-fubCfw mMVyH jzGwLo"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm-50.22 116.82C218.45 151.39 236.28 144 256 144s37.39 7.44 50.11 20.94c12.89 13.68 19.16 32.06 17.68 51.82C320.83 256 290.43 288 256 288s-64.89-32-67.79-71.25c-1.47-19.92 4.79-38.36 17.57-51.93zM256 432a175.49 175.49 0 01-126-53.22 122.91 122.91 0 0135.14-33.44C190.63 329 222.89 320 256 320s65.37 9 90.83 25.34A122.87 122.87 0 01382 378.78 175.45 175.45 0 01256 432z"
          />
        </svg>
      </div>
    </div>
  </header>
  <nav
    className="sc-pFZIQ loupFY"
  >
    <ul
      className="sc-jrAGrp bKstHo"
    >
      <li
        className="sc-kEjbxe dDbtsF"
      >
        <div
          className="sc-iqHYGH ihJcjr"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M80 212v236a16 16 0 0016 16h96V328a24 24 0 0124-24h80a24 24 0 0124 24v136h96a16 16 0 0016-16V212"
              fill="none"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="32"
            />
            <path
              d="M480 256L266.89 52c-5-5.28-16.69-5.34-21.78 0L32 256m368-77V64h-48v69"
              fill="none"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="32"
            />
          </svg>
          <small>
            Home
          </small>
        </div>
      </li>
      <li
        className="sc-kEjbxe iocGHd"
      >
        <div
          className="sc-iqHYGH ihJcjr"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M394 480a16 16 0 01-9.39-3L256 383.76 127.39 477a16 16 0 01-24.55-18.08L153 310.35 23 221.2a16 16 0 019-29.2h160.38l48.4-148.95a16 16 0 0130.44 0l48.4 149H480a16 16 0 019.05 29.2L359 310.35l50.13 148.53A16 16 0 01394 480z"
            />
          </svg>
          <small>
            Favorites
          </small>
        </div>
      </li>
      <li
        className="sc-kEjbxe iocGHd"
      >
        <div
          className="sc-iqHYGH ihJcjr"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M64 144h226.75a48 48 0 0090.5 0H448a16 16 0 000-32h-66.75a48 48 0 00-90.5 0H64a16 16 0 000 32zm384 224h-66.75a48 48 0 00-90.5 0H64a16 16 0 000 32h226.75a48 48 0 0090.5 0H448a16 16 0 000-32zm0-128H221.25a48 48 0 00-90.5 0H64a16 16 0 000 32h66.75a48 48 0 0090.5 0H448a16 16 0 000-32z"
            />
          </svg>
          <small>
            Options
          </small>
        </div>
      </li>
      <li
        className="sc-kEjbxe iocGHd"
      >
        <div
          className="sc-iqHYGH ihJcjr"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M176 176v-40a40 40 0 0140-40h208a40 40 0 0140 40v240a40 40 0 01-40 40H216a40 40 0 01-40-40v-40"
              fill="none"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="32"
            />
            <path
              d="M272 336l80-80-80-80M48 256h288"
              fill="none"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="32"
            />
          </svg>
          <small>
            Login
          </small>
        </div>
      </li>
    </ul>
  </nav>
  <main
    className="sc-crrsfI jZLqQj"
  />
</div>
`;
